src/db/transaction.rs

```diff
@@ -202,4 +218,109 @@ impl FirestoreDb {
     ) -> FirestoreResult<FirestoreTransaction> {
         FirestoreTransaction::new(self, options).await
     }
+
+    pub async fn run_transaction<T, F>(&self, func: F) -> Result<T, FirestoreError>
+    where
+        F: for<'a> Fn(
+            FirestoreDb,
+            &'a mut FirestoreTransaction,
+        )
+            -> Pin<Box<dyn futures::Future<Output = Result<T, FirestoreError>> + 'a>>,
+    {
+        let options = FirestoreTransactionOptions::default();
```

Comment:
Nit: there is `::new()` for free from `Builder` macro coming without even introducing `Default`.